core/server/common/src/test/java/alluxio/master/journal/raft/RaftJournalTest.java

```diff
@@ -242,9 +242,10 @@ public void gainPrimacyAfterSuspend() throws Exception {
     Assert.assertEquals(0, countingMaster.getApplyCount());
     // Gain primacy in follower journal and validate it catches up.
     mFollowerJournalSystem.gainPrimacy();
-    CommonUtils.waitFor("full state acquired after resume",
-        () -> countingMaster.getApplyCount() == entryCount, mWaitOptions);
-
+    CommonUtils.waitFor(
+        "full state acquired after resume", () -> mFollowerJournalSystem.getCurrentSequenceNumbers()
+            .values().stream().distinct().collect(Collectors.toList()).get(0) == entryCount - 1,
```

Comment:
old calculation was wrong?

Reply:
Gaining primacy in this test does not ensure all journal entries are applied. Sometimes a few entries are processed after the journal system is promoted and thus get ignored. The sequence number reflects entries that are processed which will include the entries processed before and after the follower becomes primary.